packages/components/src/icon/stories/index.story.tsx

```diff
@@ -32,42 +32,83 @@ Default.args = {
 	icon: wordpress,
 };
 
-export const FillColor: StoryFn< typeof Icon > = ( args ) => {
+/**
+ * When `icon` is an SVG element, the `currentColor` prop can be used to
+ * render it in the CSS `currentColor`.
+ */
+export const WithCurrentColor: StoryFn< typeof Icon > = ( args ) => {
 	return (
 		<div
 			style={ {
-				fill: 'blue',
+				background: 'blue',
+				color: 'white',
 			} }
 		>
 			<Icon { ...args } />
+			Some text
 		</div>
 	);
 };
-FillColor.args = {
+WithCurrentColor.args = {
 	...Default.args,
+	currentColor: true,
 };
 
+const renderIcon = ( { size = 24, ...restProps } ) => (
+	<div style={ { width: size } } { ...restProps }>
+		<SVG viewBox="0 0 24 24">
+			<Path d="M5 4v3h5.5v12h3V7H19V4z" />
+		</SVG>
+	</div>
+);
+
 export const WithAFunction = Template.bind( {} );
 WithAFunction.args = {
 	...Default.args,
-	icon: () => (
-		<SVG>
+	icon: renderIcon,
+};
+WithAFunction.parameters = {
+	docs: {
+		source: {
+			code: `
+const renderIcon = ( { size = 24, ...restProps } ) => (
+	<div style={ { width: size } } { ...restProps }>
+		<SVG viewBox="0 0 24 24">
 			<Path d="M5 4v3h5.5v12h3V7H19V4z" />
 		</SVG>
-	),
+	</div>
```

Comment:
I improved the code snippets for the "with function" and "with component" stories to better demonstrate how the function/components need to be implemented in order to work as intended.

(Related to #56448, where part of the problem was that we weren't documenting how these `size` props need to be handled.)